File: RanNuan TV/apps/mobile-android/app/src/main/java/com/rannuan/tv/ui/util/ImageProxy.kt
package com.rannuan.tv.ui.util

import com.rannuan.tv.BuildConfig

object ImageProxy {
    private val baseUrl get() = BuildConfig.SERVER_URL

    fun proxy(url: String?): String {
        if (url.isNullOrBlank()) return ""
        if (url.contains("/api/img")) return url
        if (url.contains("bfzy") || url.contains("picbf") ||
            url.contains("doubanio.com") || url.contains("douban.com")
        ) {
            return "$baseUrl/api/img?url=${java.net.URLEncoder.encode(url, "UTF-8")}"
        }
        return url
    }
}

File: RanNuan TV/apps/desktop/src/utils/imageProxy.ts
/**
 * 统一的图片代理函数
 */
export function proxyImg(rawUrl: string): string {
  if (!rawUrl) return '';

  // 已经是代理 URL，直接返回
  if (rawUrl.includes('/api/img')) return rawUrl;

  // 暴风资源需要代理
  if (rawUrl.includes('bfzy') || rawUrl.includes('picbf')) {
    return `http://localhost:3000/api/img?url=${encodeURIComponent(rawUrl)}`;
  }

  // 豆瓣图片防盗链，需要代理
  if (rawUrl.includes('doubanio.com') || rawUrl.includes('douban.com')) {
    return `http://localhost:3000/api/img?url=${encodeURIComponent(rawUrl)}`;
  }

  // 其他直连
  return rawUrl;
}
